From 0abb461bf75af2b70830bbdda3fea850fa5914e4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@googlemail.com>
Date: Wed, 26 Nov 2014 10:27:29 +0100
Subject: [PATCH] Add versions to resources
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

see qtwayland commit bbd4a8f85b2e0ddaa94458569b8b5aa4128c52b1
Author: Giulio Camuffo <giulio.camuffo@jollamobile.com>
Date:   Tue Sep 30 17:09:49 2014 +0300

    Fix the compositor's wayland versioning

    We cannot just assume all clients implement the same version of the
    various interfaces as the compositor does. Keep track of them, and
    react accordingly when creating a resource or sending an event.

    Change-Id: I9792433a14d49c5c4df0c892fc1349ce0dfb0d43
    Reviewed-by: Jørgen Lind <jorgen.lind@digia.com>

Signed-off-by: Andreas Müller <schnitzeltony@googlemail.com>
---
 src/compositor/protocols/plasma/plasmasurface.cpp    | 2 +-
 src/compositor/protocols/wl-shell/wlshellsurface.cpp | 2 +-
 src/compositor/protocols/xdg-shell/xdgpopup.cpp      | 2 +-
 src/compositor/protocols/xdg-shell/xdgsurface.cpp    | 2 +-
 src/compositor/quicksurface.cpp                      | 2 +-
 5 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/compositor/protocols/plasma/plasmasurface.cpp b/src/compositor/protocols/plasma/plasmasurface.cpp
index 3a09dd7..29be679 100644
--- a/src/compositor/protocols/plasma/plasmasurface.cpp
+++ b/src/compositor/protocols/plasma/plasmasurface.cpp
@@ -37,7 +37,7 @@ namespace GreenIsland {
 PlasmaSurface::PlasmaSurface(PlasmaShell *shell, QuickSurface *surface,
                              wl_client *client, uint32_t id)
     : QWaylandSurfaceInterface(surface)
-    , QtWaylandServer::org_kde_plasma_surface(client, id)
+    , QtWaylandServer::org_kde_plasma_surface(client, id, 1)
     , m_compositor(qobject_cast<Compositor *>(surface->compositor()))
     , m_shell(shell)
     , m_surface(surface)
diff --git a/src/compositor/protocols/wl-shell/wlshellsurface.cpp b/src/compositor/protocols/wl-shell/wlshellsurface.cpp
index c9f93a4..a4deb39 100644
--- a/src/compositor/protocols/wl-shell/wlshellsurface.cpp
+++ b/src/compositor/protocols/wl-shell/wlshellsurface.cpp
@@ -44,7 +44,7 @@ namespace GreenIsland {
 WlShellSurface::WlShellSurface(WlShell *shell, QuickSurface *surface,
                                wl_client *client, uint32_t id)
     : QWaylandSurfaceInterface(surface)
-    , QtWaylandServer::wl_shell_surface(client, id)
+    , QtWaylandServer::wl_shell_surface(client, id, 1)
     , m_shell(shell)
     , m_surface(surface)
     , m_moveGrabber(Q_NULLPTR)
diff --git a/src/compositor/protocols/xdg-shell/xdgpopup.cpp b/src/compositor/protocols/xdg-shell/xdgpopup.cpp
index 91b0a46..cf312fc 100644
--- a/src/compositor/protocols/xdg-shell/xdgpopup.cpp
+++ b/src/compositor/protocols/xdg-shell/xdgpopup.cpp
@@ -36,7 +36,7 @@ namespace GreenIsland {
 XdgPopup::XdgPopup(XdgShell *shell, QWaylandSurface *parent, QWaylandSurface *surface,
                    wl_client *client, uint32_t id, uint32_t serial)
     : QWaylandSurfaceInterface(surface)
-    , QtWaylandServer::xdg_popup(client, id)
+    , QtWaylandServer::xdg_popup(client, id, 1)
     , m_shell(shell)
     , m_parentSurface(parent)
     , m_surface(surface)
diff --git a/src/compositor/protocols/xdg-shell/xdgsurface.cpp b/src/compositor/protocols/xdg-shell/xdgsurface.cpp
index 62783b1..15ff8a2 100644
--- a/src/compositor/protocols/xdg-shell/xdgsurface.cpp
+++ b/src/compositor/protocols/xdg-shell/xdgsurface.cpp
@@ -44,7 +44,7 @@ namespace GreenIsland {
 XdgSurface::XdgSurface(XdgShell *shell, QuickSurface *surface,
                        wl_client *client, uint32_t id)
     : QWaylandSurfaceInterface(surface)
-    , QtWaylandServer::xdg_surface(client, id)
+    , QtWaylandServer::xdg_surface(client, id, 1)
     , m_shell(shell)
     , m_surface(surface)
     , m_moveGrabber(Q_NULLPTR)
diff --git a/src/compositor/quicksurface.cpp b/src/compositor/quicksurface.cpp
index 8de2f24..d24db01 100644
--- a/src/compositor/quicksurface.cpp
+++ b/src/compositor/quicksurface.cpp
@@ -30,7 +30,7 @@
 namespace GreenIsland {
 
 QuickSurface::QuickSurface(wl_client *client, quint32 id, Compositor *compositor)
-    : QWaylandQuickSurface(client, id, compositor)
+    : QWaylandQuickSurface(client, id, 2, compositor)
     , m_state(Normal)
     , m_globalPos(-1, -1)
 {
-- 
1.8.3.1

